### PR TITLE
[tests-only] Adds e2e tests for managing space admin settings via context menu

### PR DESCRIPTION
**Pull request by SwikritiT**
## Description
This PR extends the tests to be able to edit space changing following using a context menu
- rename project space
- edit quota of a project space
- enable project space

## Related Issue

- Part of: https://github.com/owncloud/web/issues/8601

## How Has This Been Tested?

- locally 
- ci

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

File: tests/e2e/cucumber/features/smoke/admin-settings/spaces.ocis.feature [hunk 1]
```diff
@@ -19,20 +19,28 @@ Feature: spaces management
     Then "Alice" should see the following spaces
       | id     |
       | team.a |
-    And "Alice" updates the quota for space "team.a" to "50"
+    When "Alice" updates the space "team.a" name to "developer team" using the context-menu
+    And "Alice" updates the space "team.a" subtitle to "developer team-subtitle" using the context-menu
+    And "Alice" updates the space "team.a" quota to "50" using the context-menu
+    And "Alice" disables the space "team.a" using the context-menu
+    And "Alice" enables the space "team.a" using the context-menu
```
Comment (ScharfViktor) [inline tests/e2e/cucumber/features/smoke/admin-settings/spaces.ocis.feature:26]: I would suggest to reduce repeated steps:
for example: 
for space team.a we can
- update name
- update subtitle
- update quota
- disable
- enable
- should see team.a

for space team.b:
- disable and delete
- should not see team.b

What do you thing? 
or for space team.b -> updates atribytes by disabled space

File: tests/e2e/cucumber/features/smoke/admin-settings/spaces.ocis.feature [hunk 2]
```diff
@@ -19,20 +19,28 @@ Feature: spaces management
     Then "Alice" should see the following spaces
```
Comment (ScharfViktor) [inline tests/e2e/cucumber/features/smoke/admin-settings/spaces.ocis.feature:19]: maybe remove this step?

File: tests/e2e/support/objects/app-admin-settings/spaces/actions.ts [hunk 3]
```diff
@@ -8,6 +8,11 @@ const spaceCheckboxSelector = `[data-item-id="%s"]:not(.oc-table-highlighted) in
 const quotaActionBtn = `.oc-files-actions-edit-quota-trigger`
 const disableActionBtn = `.oc-files-actions-disable-trigger`
 const deleteActionBtn = `.oc-files-actions-delete-trigger`
+const enableActionBtn = `.oc-files-actions-restore-trigger`
+const renameActionBtn = `.oc-files-actions-rename-trigger`
+const editSubtitleActionBtn = `.oc-files-actions-edit-description-trigger`
```
Comment (ScharfViktor) [inline tests/e2e/support/objects/app-admin-settings/spaces/actions.ts:13]: maybe use as:
`const actionBtn = '.oc-files-actions-%s-trigger'`

and after that use one button: 
`locator(util.format(actionBtn, 'edit-description')`

File: tests/e2e/support/objects/app-admin-settings/spaces/actions.ts [hunk 4]
```diff
@@ -77,6 +82,32 @@ export const disableSpace = async (args: {
   ])
 }
 
+export const enableSpace = async (args: {
```
Comment (ScharfViktor) [inline tests/e2e/support/objects/app-admin-settings/spaces/actions.ts:85]: these methods(enableSpace, disable, delete, change) are very similar. should they be combined into one? 

But that isn't request to change it :slight_smile:

File: tests/e2e/cucumber/features/smoke/admin-settings/spaces.ocis.feature (hunk 1 above)
Comment (SwikritiT) [reply tests/e2e/cucumber/features/smoke/admin-settings/spaces.ocis.feature:26]: yes that sounds good. I'll update them accordingly

File: tests/e2e/cucumber/features/smoke/admin-settings/spaces.ocis.feature [hunk 5]
```diff
@@ -16,23 +16,21 @@ Feature: spaces management
...
     And "Alice" logs out
-    Then "Brian" logs in
+    When "Brian" logs in
     And "Brian" opens the "admin-settings" app
     And "Brian" navigates to the project spaces management page
     Then "Brian" should see the following spaces
       | id     |
       | team.b |
-    And "Brian" updates the quota for space "team.b" to "50"
     And "Brian" disables the space "team.b" using the context-menu
```
Comment (amrita-shrestha) [inline tests/e2e/cucumber/features/smoke/admin-settings/spaces.ocis.feature:34]:
```suggestion
    When "Brian" disables the space "team.b" using the context-menu
```

File: tests/e2e/support/objects/app-admin-settings/spaces/actions.ts (hunk 4 above)
Comment (amrita-shrestha) [reply tests/e2e/support/objects/app-admin-settings/spaces/actions.ts:85]: I think we can DRY code here or can't we @SwikritiT

File: tests/e2e/support/objects/app-admin-settings/spaces/index.ts [hunk 6]
```diff
@@ -36,6 +39,10 @@ export class Spaces {
     await disableSpace({ id, page: this.#page, context })
   }
 
+  async enable({ key, context }: { key: string; context: string }): Promise<void> {
+    const { id } = this.#spacesEnvironment.getSpace({ key })
+    await enableSpace({ id, page: this.#page, context })
+  }
```
Comment (amrita-shrestha) [inline tests/e2e/support/objects/app-admin-settings/spaces/index.ts:45]:
```suggestion
  }
  
```

File: tests/e2e/support/objects/app-admin-settings/spaces/actions.ts (hunk 3 above)
Comment (SwikritiT) [reply tests/e2e/support/objects/app-admin-settings/spaces/actions.ts:13]: @ScharfViktor I'll do it in this PR https://github.com/owncloud/web/pull/8629/files

File: tests/e2e/support/objects/app-admin-settings/spaces/actions.ts (hunk 4 above)
Comment (SwikritiT) [reply tests/e2e/support/objects/app-admin-settings/spaces/actions.ts:85]: @amrita-shrestha @ScharfViktor  yes it's nice to make code DRY, will be in this PR https://github.com/owncloud/web/pull/8629/files

**Review by ScharfViktor — APPROVED** [review]
LGTM

**Pull request merged**